src/hotspot/cpu/aarch64/aarch64_sve.ad

```diff
@@ -4078,3 +4084,32 @@ instruct rearrangeD(vReg dst, vReg src, vReg shuffle)
   %}
   ins_pipe(pipe_slow);
 %}
+
+// ------------------------------ Vector Load Gather ---------------------------------
+instruct gatherI(vReg dst, vmemA mem, vReg idx) %{
+  predicate(UseSVE > 0 &&
+           (n->bottom_type()->is_vect()->element_basic_type() == T_INT ||
+            n->bottom_type()->is_vect()->element_basic_type() == T_FLOAT));
+  match(Set dst (LoadVectorGather mem idx));
+  ins_cost(SVE_COST);
+  format %{ "load_vector_gather $dst, $mem, $idx\t# vector load gather (I/F)" %}
+  ins_encode %{
+    __ sve_ld1w_gather(as_FloatRegister($dst$$reg), ptrue, as_Register($mem$$base), as_FloatRegister($idx$$reg));
+  %}
+  ins_pipe(pipe_slow);
+%}
+
+instruct gatherL(vReg dst, vmemA mem, vReg idx) %{
+  predicate(UseSVE > 0 &&
+           (n->bottom_type()->is_vect()->element_basic_type() == T_LONG ||
+            n->bottom_type()->is_vect()->element_basic_type() == T_DOUBLE));
+  match(Set dst (LoadVectorGather mem idx));
+  ins_cost(2 * SVE_COST);
+  format %{ "sve_uunpklo $idx, $idx\n\t"
+            "load_vector_gather $dst, $mem, $idx\t# vector load gather (L/D)" %}
+  ins_encode %{
+    __ sve_uunpklo(as_FloatRegister($idx$$reg), __ D, as_FloatRegister($idx$$reg));
+    __ sve_ld1d_gather(as_FloatRegister($dst$$reg), ptrue, as_Register($mem$$base), as_FloatRegister($idx$$reg));
+  %}
+  ins_pipe(pipe_slow);
+%}
```

Comment:
Thanks for your contribution! This patch looks good to me. I just have one question: since currently the SVE supports the vector operations for vector length smaller than the max vector size, does it need to  consider the partial implementation for the `LoadVectorGather `op? Thanks!

Reply:
Thank you for your review. In my opinion, partial version of `LoadVectorGather` is useful. However, I don't know the plan of `mask`  work of SVE, so I have no plan to implement that now.

Reply:
Actually this does not strongly depends on the mask feature support for SVE. And we already have the partial implementation that are merged into the panama-vector. I think you can have them as the reference. Thanks!

Reply:
> Actually this does not strongly depends on the mask feature support for SVE. And we already have the partial implementation that are merged into the panama-vector. I think you can have them as the reference. Thanks!

OK. I will check that after implementation of all missing nodes. Thank you.

Reply:
OK, A a follow-up patch also makes sense to me. Thanks!